source/common/upstream/cluster_manager_impl.h

```diff
@@ -857,19 +883,15 @@ class ClusterManagerImpl : public ClusterManager,
 
   bool deferralIsSupportedForCluster(const ClusterInfoConstSharedPtr& info) const;
 
+  const envoy::config::bootstrap::v3::Bootstrap& bootstrap_;
```

Comment:
I should probably know this, but is the bootstrap object lifetime longer than the cluster-manager initialization? (I'm worried about non `init()` methods using this bootstrap).
Consider passing the bootstrap as a parameter to the `init()` method.